_docs/technical/2017-01-04-updater.md

```diff
@@ -93,14 +99,17 @@ As nodes deserialize
 they modify mempool as implemented
 [here](https://github.com/input-output-hk/cardano-sl/blob/22360aa45e5dd82d0c87872d8530217fc3d08f4a/src/Pos/Update/MemState/Functions.hs#L40).
 
+[//]: TODO: what to say about votes?
```

Comment:
I don't understand why `Updating the Mempool` is a subsection of `Proposal Accumulation`.
Proposal accumulation is a particular mempool update. There are other updates, e. g. votes accumulation.

`MemPool` consists of votes and proposals. Apart from that `MemState` contains tip, slot and `PollModifier` corresponding to `MemPool` (and to current GState, i. e. to application of `MemPool` to GState).
It is updated in three cases:
1. When new proposal/vote is received. In this case one of the `process*` functions from `Pos.Update.Logic.Local` is called which in turn calls `verifyAndApplyUSPayload` and then updates current `PollModifier` and `MemPool`.
2. When new slot starts. In this case some data in mempool may become invalid. In fact, it happens only when epoch changes. That's because stable stake distribution changes and some votes may have not enough stake for inclusion, for example. It's done in `processNewSlot` function.
3. When GState is updated. It is called `usNormalize`. Some data may become invalid as a result of block(s) application or rollback. For instance, we have proposal in memory, apply block with this proposal and it becomes invalid (because it's already in block). We should drop such proposal. Or we have vote for proposal from some block, then rollback of this block happens and vote becomes no longer valid. It is implemented by applying all local data to empty state ignoring all data which is no longer valid.

I didn't intend to write something that can be put into documentation verbatim, but I hope it will be useful.